src/NixFromNpm/Npm/Version.hs

```diff
@@ -86,3 +86,37 @@ instance FromJSON NpmVersionRange where
       Nothing -> return $ InvalidVersion s
       Just range -> return range
     _ -> Aeson.typeMismatch "string" v
+
+-- | A package name can be passed in directly, or a version range can be
+-- specified with a %.
+parseNameAndRange :: MonadIO m => Text -> m (PackageName, NpmVersionRange)
+parseNameAndRange name = do
+  let badFormat err = UnrecognizedVersionFormat (name <> " (" <> err <> ")")
+
+  let parseName n = case parsePackageName n of
+        Left err -> throw $ badFormat err
```

Comment:
Non-blocking comment. It might be nice to refactor this in the future to use ExceptT instead of throwing an exception.

Reply:
Definitely; it was just the easiest way to break stuff up :)